bootstrap/postgres/configs.go

```diff
@@ -451,59 +460,86 @@ func (cr configRepository) RemoveChannel(ctx context.Context, id string) error {
 	return nil
 }
 
+func (cr configRepository) ConnectThing(ctx context.Context, channelID, thingID string) error {
+	q := `UPDATE configs SET state = $1 WHERE EXISTS (
+	SELECT 1 FROM connections WHERE config_id = $2 AND channel_id = $3)`
+	result, err := cr.db.ExecContext(ctx, q, bootstrap.Active, thingID, channelID)
+	if err != nil {
+		return errors.Wrap(errConnectThing, err)
+	}
+	if rows, _ := result.RowsAffected(); rows == 0 {
+		return repoerr.ErrNotFound
+	}
+	return nil
+}
+
 func (cr configRepository) DisconnectThing(ctx context.Context, channelID, thingID string) error {
 	q := `UPDATE configs SET state = $1 WHERE EXISTS (
 		SELECT 1 FROM connections WHERE config_id = $2 AND channel_id = $3)`
-	if _, err := cr.db.ExecContext(ctx, q, bootstrap.Inactive, thingID, channelID); err != nil {
+	result, err := cr.db.ExecContext(ctx, q, bootstrap.Inactive, thingID, channelID)
+	if err != nil {
 		return errors.Wrap(errDisconnectThing, err)
 	}
+	if rows, _ := result.RowsAffected(); rows == 0 {
+		return repoerr.ErrNotFound
+	}
 	return nil
 }
 
-func (cr configRepository) retrieveAll(owner string, filter bootstrap.Filter) (string, []interface{}) {
-	template := `WHERE owner = $1 %s`
-	params := []interface{}{owner}
-	// One empty string so that strings Join works if only one filter is applied.
-	queries := []string{""}
-	// Since owner is the first param, start from 2.
-	counter := 2
+func (cr configRepository) retrieveAll(domainID string, thingIDs []string, filter bootstrap.Filter) (string, []interface{}) {
+	params := []interface{}{}
+	queries := []string{}
+
+	if domainID != "" {
+		params = append(params, domainID)
+		queries = append(queries, fmt.Sprintf("domain_id = $%d", len(params)))
+	}
```

Comment:
Why we made domainID as optional here ? 
In any case domainID might be empty?

Reply:
I made `domainID` optional in the `retrieveAll` function to allow for more flexible querying.

Reply:
But If domainID becomes empty, then it should not return any config, 
having a test case for this will be good.